AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/HierarchicalVersionUtils.dfy

```diff
@@ -0,0 +1,128 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+include "../Model/AwsCryptographyKeyStoreTypes.dfy"
+include "Structure.dfy"
+include "KMSKeystoreOperations.dfy"
+include "KmsArn.dfy"
+
+module HierarchicalVersionUtils {
+
+  import opened StandardLibrary
+  import opened Wrappers
+  import opened Seq
+
+  import ErrorMessages = KeyStoreErrorMessages
+  import Types = AwsCryptographyKeyStoreTypes
+  import KMS = ComAmazonawsKmsTypes
+  
+  import AtomicPrimitives
+  import UTF8
+  import KMSKeystoreOperations
+  import KmsArn
+  import Structure
+  
+  function method GetMdDigestFromEC(
+    item: Types.EncryptionContextString
+  ) : (output: Types.EncryptionContextString)
+  ensures output.Keys == item.Keys - {Structure.TABLE_FIELD}
+  ensures forall k :: k in output ==> output[k] == item[k]
+  ensures forall k :: k in output ==> k !in {Structure.TABLE_FIELD}
+  {
+    map k <- item.Keys - {Structure.TABLE_FIELD}
+      :: k := item[k]
+  }
+  
+  method GetHV2EC(
```

Comment:
I get it, but for today, let's distinguish b/w the various ECs.
KMS EC :: What KMS sees.
```suggestion
  method GetHv2KmsEc(
```